### PR TITLE
Cleanup for filters optimizations

### PR DESCRIPTION
**Pull request by nikhilunni**
## Change description

* Added some basic tests + test infra to help us
* Addressed comments from last PR that I hastily merged

## Type of change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Improvement (improves codebase without changing functionality)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,9 +61,6 @@
     "@backstage/test-utils": "^1.0.0",
     "@cortexapps/eslint-config-oss": "^0.0.3",
     "@spotify/prettier-config": "^11.0.0",
-    "@testing-library/jest-dom": "^5.10.1",
```
Comment (nikhilunni) [inline package.json:64]: these were all causing version conflicts with backstage testing util libraries

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardDetailsPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+  function render() {
+    return renderWrapped(<ScorecardDetailsPage />, {
+      '/': rootRouteRef,
+      '/scorecards/:id': scorecardRouteRef as any,
+    }, cortexApi,[catalogApiRef, catalogApi], [extensionApiRef, emptyExtensionApi]);
+  }
+
+
+  it('should be render all service scores', async () => {
```
Comment (FoodProduct) [inline src/components/Scorecards/ScorecardDetailsPage/ScorecardDetailsPage.test.tsx:95]:
```suggestion
  it('should render all service scores', async () => {
```

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardDetailsPage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    await checkForText('bar');
+    await clickButton('Filter groups by shared')
+
+    await clickButton('Filter groups by alsomine')
+    await checkNotText('foo');
+    await checkForText('bar');
+    await clickButton('Filter groups by alsomine')
+  });
+
+  // MUI select is broken in a weird way, can't test
```
Comment (FoodProduct) [inline src/components/Scorecards/ScorecardDetailsPage/ScorecardDetailsPage.test.tsx:125]: I ran into similar issues 😞

File: src/utils/hooks.ts [hunk 4]
```diff
@@ -104,15 +104,15 @@ export function useGroupsAndSystems(): GroupAndSystemsHookType {
...
+        stringifyAnyEntityRef(entityRef, { defaultKind: 'system' }),
       );
     });
 
     const groups = groupByString(entities.items, entity => {
       return getEntityRelations(entity, RELATION_OWNED_BY, {
         kind: 'group',
       }).map(entityRef =>
-        stringifyAnyEntityRef(entityRef, { defaultKind: 'system' }),
+        stringifyAnyEntityRef(entityRef, { defaultKind: 'group' }),
```
Comment (FoodProduct) [inline src/utils/hooks.ts:115]: 😬 Eep good catch!

**Review by FoodProduct — APPROVED** [review]
Awesome cleanup; thank you! 🙇

**Review by FoodProduct — APPROVED** [review]

**Pull request merged**